From 62961d640e3d751e594ebe6d7388ce026165df3a Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 12 Oct 2016 14:53:28 +0800
Subject: [PATCH 1199/5242] MLK-12928-16 regulator: pf1550: check device node
 check

commit  d0430a4914934561551da66043589a9a900a0208 from
https://source.codeaurora.org/external/imx/linux-imx.git

Do not probe if the device node is not correct in dts.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/pf1550.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/regulator/pf1550.c b/drivers/regulator/pf1550.c
index c7e8bfb..f683705 100644
--- a/drivers/regulator/pf1550.c
+++ b/drivers/regulator/pf1550.c
@@ -287,10 +287,14 @@ static irqreturn_t pf1550_regulator_irq_handler(int irq, void *data)
 static int pf1550_regulator_probe(struct platform_device *pdev)
 {
 	struct pf1550_dev *iodev = dev_get_drvdata(pdev->dev.parent);
+	struct device_node *np = pdev->dev.of_node;
 	struct pf1550_regulator_info *info;
 	int i, ret = 0;
 	struct regulator_config config = { };
 
+	if (!np)
+		return -ENODEV;
+
 	info = devm_kzalloc(&pdev->dev, sizeof(struct pf1550_regulator_info),
 				   GFP_KERNEL);
 	if (!info)
-- 
1.7.9.5

